src/main/java/dk/cngroup/lentils/controller/TeamController.java

```diff
@@ -2,71 +2,80 @@
 
 import dk.cngroup.lentils.entity.Team;
 import dk.cngroup.lentils.service.TeamService;
+import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
 import org.springframework.validation.BindingResult;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.validation.Errors;
+import org.springframework.web.bind.annotation.*;
 
 import javax.validation.Valid;
+import java.io.IOException;
+import java.util.InputMismatchException;
+import java.util.List;
 
 @Controller
 @RequestMapping("/team")
 public class TeamController {
     private final TeamService teamService;
 
-    private final String VIEW_PATH = "team/";
+    private final String VIEW_PATH = "team/list";
+    private final String REDIRECT = "redirect:/team/add";
+    private final String ACTION_TEAM_ADD = "/team/add";
+    private final String ACTION_TEAM_UPDATE = "/team/update/";
 
     public TeamController(TeamService teamService) {
         this.teamService = teamService;
     }
 
     @GetMapping(value = "/add")
     public String addTeam(Model model) {
-        model.addAttribute("teams", teamService.getAll());
-        model.addAttribute("team", new Team());
-        model.addAttribute("action", "/team/add");
-        return VIEW_PATH + "list";
+        fillModelAttributes(model, teamService.getAll(), new Team(), ACTION_TEAM_ADD);
+        return VIEW_PATH ;
     }
 
     @PostMapping(value = "/add")
     public String addTeam(@Valid Team team, Model model, BindingResult bindingResult) {
         if (bindingResult.hasErrors()) {
-            model.addAttribute("teams", teamService.getAll());
-            model.addAttribute("team", new Team());
-            model.addAttribute("action", "/team/add");
-            return VIEW_PATH + "list";
+            fillModelAttributes(model, teamService.getAll(), new Team(), ACTION_TEAM_ADD);
+            return VIEW_PATH ;
         }
         teamService.add(team);
-        return "redirect:/team/add";
+        return REDIRECT;
+    }
+
+    @ExceptionHandler(InputMismatchException.class)
+    @ResponseStatus(HttpStatus.NOT_FOUND)
+    public Error handleNotFoundException(InputMismatchException e)
+    {
+        return new Error(e.getMessage());
     }
 
     @GetMapping("/update/{id}")
     public String update(@PathVariable("id") Long id, Model model) {
```

Comment:
Me by se asi cela ta metoda libila vic takto:
```
public String update(@PathVariable("id") Long id, Model model) {
	List<Team> teams = teamService.getAll();
	Team team = teamService.get(id);
	String action = ACTION_TEAM_UPDATE + id;
	fillModelAttributes(model, teams, team, action);

	return VIEW_PATH;
}
```
A nebo to nechat tak, jak to bylo. Je dobre, ze si odstranila duplicitu tou metodou fillModelAttributes, ale kdyz jsou ty parametry takto poslane v jednom radku, tak neni uplne jasne, ktery parametr co reprezentuje.